### DISCUSSION TITLE
Exercício javascript

### DISCUSSION THREAD

**Discussion by lucastorresbr** [frontendbr/forum#1868] [Perguntas e Respostas]
Galera, tenho um exercício que diz o seguinte:

Maior que o número
Programe uma função maiorQueNum que busca em um dado array apenas os números maiores do que o número fornecido no segundo parâmetro da função e retorne um novo array apenas com esses números.
Exemplo:
numeros = [10, 4, 7, 128, 42, -1, 0, 300, -5]
num = 5

maiorQueNum(numeros, num); // retorna [10, 7, 128, 42, 300]

Como faço isso? Sei que tenho que usar condicionais na função e tbm o return.

**Comment by brendomorassi** [comment]
Está aqui, legal a pegadinha do 0 não sei se a proposta era essa, mas atendeu kkkkk.

`var numeros = [10, 4, 7, 128, 42, -1, 0, 300, -5];
var num = 5

function maiorQueNum(numeros, num){
var newArray =[]; 
for(var i=0; i < numeros.length; i++){
if(numeros[i] > num){
newArray.push(numeros[i]);
}
}
 return newArray
}

console.log(maiorQueNum(numeros,num));`

**Comment by igorthurow** [comment]
Oi amigo! Tudo bem?

O método de Array [filter](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array/filtro) vai te cair muito bem!
Inclusive se achar interessante dar uma olhada na sessão de métodos de array [aqui](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/Array), talvez possa incrementar suas ferramentas pra resolver esses problemas. 

Sucesso!

```
const numbers = [10, 4, 7, 128, 42, -1, 0, 300, -5]
const numberValidation = 5

const biggerThanNumber = (numbers, numberValidation) => (
   numbers.filter((number) => number > numberValidation)
)

biggerThanNumber(numbers, numberValidation) // [10, 7, 128, 42, 300]
```
